sdk/aqueduct/flow_run.py

```diff
@@ -64,6 +68,30 @@ def describe(self) -> None:
             assert param_op.spec.param is not None, "Artifact is not a parameter."
             print("* " + param_op.name + ": " + param_op.spec.param.val)
 
+    def artifact(
+        self, artifact_name: str
+    ) -> Union[TableArtifact, MetricArtifact, CheckArtifact, ParamArtifact, None]:
+        """Gets the Artifact from the flow based on artifact_name"""
+        flow_run_dag = self._dag
+
+        for artifact_from_workflow in list(flow_run_dag.artifacts.values()):
+            if artifact_from_workflow.name == artifact_name:
+                assert isinstance(artifact_from_workflow, Artifact)
+                artifact_DAG = artifact_from_workflow
```

Comment:
this fetching of the artifact from the dag should be a method on the dag class (see dag.py). You can either augment `must_get_artifact()` or write a separate method called `get_artifact_by_name()` or something - see how all the other functions on this class do it. You'll probably have to define an `artifact_by_name` dictionary to do the lookup (similar to how `operator_by_name` works).

It should be simple here as:
```
artifact = self._dag.must_get_artifact(name)
```